File: src/vs2012/RibbonItemTemplateVB/MyRibbon.vb
Imports SolidEdgeCommunity.AddIn
Imports SolidEdgeCommunity.Extensions ' https://github.com/SolidEdgeCommunity/SolidEdge.Community/wiki/Using-Extension-Methods

Public Class $safeitemname$
    Inherits SolidEdgeCommunity.AddIn.Ribbon

    Private _button1 As RibbonButton

    Public Sub New()
        ' Get a reference to the current assembly. This is where the ribbon XML is embedded.
        Dim assembly = System.Reflection.Assembly.GetExecutingAssembly()

        ' In this example, XML file must have a build action of "Embedded Resource".
        Dim embeddedResourceName = String.Format("{0}.xml", Me.GetType().FullName)
        Me.LoadXml(assembly, embeddedResourceName)

        ' Example of how to bind a local variable to a particular ribbon control.
        _button1 = GetButton(0)

        ' Example of how to bind a particular ribbon control click event.
        AddHandler _button1.Click, AddressOf _button1_Click
    End Sub

    Public Overrides Sub OnControlClick(ByVal control As RibbonControl)
        Dim application = MyAddIn.Instance.Application

        ' Demonstrate how to handle commands without binding to a local variable.
        Select Case control.CommandId
            Case 0
        End Select
    End Sub

    Private Sub _button1_Click(ByVal control As RibbonControl)
    End Sub
End Class

File: src/vs2012/RibbonItemTemplateCS/MyRibbon.cs
﻿using SolidEdgeCommunity.AddIn;
using SolidEdgeCommunity.Extensions; // https://github.com/SolidEdgeCommunity/SolidEdge.Community/wiki/Using-Extension-Methods
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace $rootnamespace$
{
	class $safeitemrootname$ : SolidEdgeCommunity.AddIn.Ribbon
	{
        private RibbonButton _button1;

        public $safeitemname$()
        {
            // Get a reference to the current assembly. This is where the ribbon XML is embedded.
            var assembly = System.Reflection.Assembly.GetExecutingAssembly();

            // In this example, XML file must have a build action of "Embedded Resource".
            var _embeddedResourceName = String.Format("{0}.xml", this.GetType().FullName);
            this.LoadXml(assembly, _embeddedResourceName);

            // Example of how to bind a local variable to a particular ribbon control.
            _button1 = GetButton(0);

            // Example of how to bind a particular ribbon control click event.
            _button1.Click += _button1_Click;
        }

        public override void OnControlClick(RibbonControl control)
        {
            var application = MyAddIn.Instance.Application;

            // Demonstrate how to handle commands without binding to a local variable.
            switch (control.CommandId)
            {
                case 0:
                    break;
            }
        }

        void _button1_Click(RibbonControl control)
        {
        }
	}
}
